app/src/main/java/com/example/datarsd1/mdbsocials_2/AddSocial.java

```diff
@@ -0,0 +1,124 @@
+package com.example.datarsd1.mdbsocials_2;
+
+import android.content.ContentValues;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.net.Uri;
+import android.provider.MediaStore;
+import android.support.v7.app.AlertDialog;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ImageView;
+import android.widget.Toast;
+
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.storage.FirebaseStorage;
+import com.google.firebase.storage.StorageReference;
+
+public class AddSocial extends AppCompatActivity implements View.OnClickListener{
+
+    private EditText newTitleText, newDateText, newDescText;
+    private FirebaseAuth mAuth;
+    private String firebasePathLocation;
+    private DatabaseReference mDatabase;
+    private ImageView preview;
+
+    private Button addEventButton, uploadImageButton;
+
+    private Uri eventimageuri;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState)
+    {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_add_social);
+
+        newTitleText = (EditText)findViewById(R.id.newEventTitle);
+        newDateText = (EditText)findViewById(R.id.newEventDate);
+        newDescText = (EditText)findViewById(R.id.newEventDesc);
+
+        mDatabase = FirebaseDatabase.getInstance().getReference();
+        final String id = mDatabase.child("Socials").push().getKey();
+        mAuth = FirebaseAuth.getInstance();
+
+        preview = (ImageView)findViewById(R.id.eventImagePreview);
+
+        addEventButton = (Button)findViewById(R.id.newEventCreate);
+        uploadImageButton = (Button)findViewById(R.id.newEventImage);
+        addEventButton.setOnClickListener(this);
+        uploadImageButton.setOnClickListener(this);
+    }
+
+    public void onClick(View v)
+    {
+        switch(v.getId())

```

Comment:
For each case, instead of having long blocks of code, you could put functions instead. For example, you could have createNewEventClicked(), imageClicked(), etc. This makes the switch easier to read and modularizes the code.